### DISCUSSION TITLE
Infinity datasource - Nested Json

### DISCUSSION THREAD

**Discussion by Suntosh378** [grafana/grafana-infinity-datasource#462] [Q&A] [answered]
Hi Team,

First of all, I would like to thank the entire team of Infinity data source it's really awesome! 

I have a question about how to read or add columns of a nested Json, here is example data

`"tenantUtilizations" : {
    "items" : [ {
      "tenantName" : "default",
      "cpuUtilizationPercentage" : 0.5634362981576545,
      "memoryUtilizationPercentage" : 0.13986156525203003
    }, {
      "tenantName" : "root.test",
      "cpuUtilizationPercentage" : 3.704417759171532E-6,
      "memoryUtilizationPercentage" : 1.1931708778139904E-7
    }, {
      "tenantName" : "root.test2",
      "cpuUtilizationPercentage" : 0.28167128256512186,
      "memoryUtilizationPercentage" : 0.25575187355477363
    }, {
      "tenantName" : "root.test3"
    }, {
      "tenantName" : "root.test4",
      "cpuUtilizationPercentage" : 0.006432982429397009,
      "memoryUtilizationPercentage" : 6.211428140708E-4
    } ]
`

out of above json data I'd like to pull the metrics in the form of the table something like

tenant name     cpuUtilizationPercentage   memoryUtilizationPercentage
 default             0.5634362981576545         0.13986156525203003
root.test           3.704417759171532E-6      1.1931708778139904E-7
.
.
.
and so on

if someone share the syntax on how to parse the json data that would be really helpful and I'm pulling that data from a URL/jmx

Thank you

**Comment by yesoreyeram** [comment, marked as answer]
use `tenantUtilizations.items` as the root selector.

<img width="1791" alt="image" src="https://user-images.githubusercontent.com/153843/201864638-d197af50-7124-4cd2-a46a-e9e38052d059.png">

**Comment by Suntosh378** [comment]
Thank you Sriram, that worked like a charm. Just curious root.test3 doesn't contain any data but how to populate that with "null" values ?

**Reply by yesoreyeram** [reply]
Explore the docs

**Reply by Suntosh378** [reply]
ok, thank you
